tests/test_compiler_entrypoints.py

```diff
@@ -74,6 +76,63 @@ def verify_anchor(flnr, s, e, tolnr):
     verify_anchor(23, 22, 23, 19)
 
 
+def test_anchors_basic_old(snippetcompiler):
+    """
+    this test verify that the old path to generate the Anchormap still works. This ensure that we remain
+    backward compatible with old Language servers.
+    """
+    snippetcompiler.setup_for_snippet(
+        """
+entity Test:
+    string a = "a"
+    string b
+end
+
+entity Test2 extends Test:
+    foo c
+end
+
+Test.relation [0:1] -- Test2.reverse [0:]
+
+typedef foo as string matching /^a+$/
+
+a = Test(b="xx")
+z = a.relation
+u = a.b
+
+implementation a for Test:
+
+end
+
+implement Test using a
+""",
+        autostd=False,
+    )
+    compiler = Compiler()
+    (statements, blocks) = compiler.compile()
+    sched = scheduler.Scheduler()
+    anchormap = sched.anchormap(compiler, statements, blocks)
+
+    assert len(anchormap) == 9
+
+    checkmap = {(r.lnr, r.start_char, r.end_char): t.lnr for r, t in anchormap}
```

Comment:
anchormap returns something of type `Sequence[Tuple[Location, Location]]`, but `Location` doesn't the fields `start_char` and `end_char`. Am I missing something here?

Reply:
`Range` inherits from a `Location` and range has those fields.
The typing between `Location` and `Range` is a bit broken and not super nice but it would be to much of an effort to fix it now:

- the resolve function [here](https://github.com/inmanta/inmanta-core/blob/db32611489e71e1064d1c60df7b963200fd31114/src/inmanta/execute/scheduler.py#L283) use to return a Range or a Location before I made any change. Now it returns an AnchorTarget where its location field is a `Location` or a `Range`.

Reply:
Ok. So in this specific case the map happens to only contain elements of type `Range`. In that case we need to add an assert statement that verifies this to make clear to the reader of the remainder of the code assumes that all the elements for the `Range` type.